slick/src/main/scala/slick/compiler/ForceOuterBinds.scala

```diff
@@ -54,3 +61,17 @@ class ForceOuterBinds extends Phase {
     case _ => nowrap(n)
   }
 }
+
+class ResolveHeadOption extends Phase {
+  val name = "resolveHeadOption"
+
+  def apply(state: CompilerState): CompilerState = state.map(transformTree)
+
+  def transformTree(node: Node): Node = mapNode(node.mapChildren(transformTree)).infer()
+
+  def mapNode(node: Node) = node match {
+    case n @ HeadOption(ch) =>
+      Library.SilentCast.typed(n.nodeType, Take(ch, LiteralNode(1)).infer())
+    case other => other
+  }
+}
```

Comment:
I dont know if that phase will stay. Its a placebo for now. Once I get the thing completed with all the types resolving completely, I will be more sure on that!